soc: Kconfig: Add BPROT and MPU to list of supported HW.

hakonfam opened the pull request:
soc: Kconfig: Add BPROT and MPU to list of supported HW.
Add configuration values for HAS_HW_NRF_BPROT and HAS_HW_NRF_MPU.
Select these configurations where applicable.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

hakonfam committed:
soc: nordic: Add BPROT and MPU to list of supported HW.

Add configuration values for HAS_HW_NRF_BPROT and HAS_HW_NRF_MPU.
Select these configurations where applicable.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

ioannisg commented:
Some comments
- AFAICS, symbol HAS_HW_NRF_BPROT is not defined, so this build will fail abyways
- HAS_HW_NRF_BPROT could also be selected for nRF51
- ideally, the symbol should be added together with some piece of code that makes use of it.

ioannisg reviewed (commented):
Don't see the immediate use this patch. Please, elaborate :)

hakonfam commented:
> * AFAICS, symbol HAS_HW_NRF_BPROT is not defined, so this build will fail abyways
Fixed
> * HAS_HW_NRF_BPROT could also be selected for nRF51
It is called MPU for nRF51, added that. 

> * ideally, the symbol should be added together with some piece of code that makes use of it.
I don't have that at the moment. But it will be used when some code wants to lock down flash on a nRF device, and needs to know what peripheral it should use to lock.

anangl commented:
What's the purpose of having these symbols, how are they supposed to be used?

SebastianBoe commented:
> What's the purpose of having these symbols, how are they supposed to be used?

The purpose is to allow customer-developed out-of-tree BPROT/MPU drivers to depend on HAS_<peripheral>.

carlescufi commented:
@ioannisg @anangl please re-review

anangl reviewed (commented):
Looks good.
I'd only suggest adding "nordic:" in the commit title so that it is visible at first glance which SoCs this commit refers to.

hakonfam commented:
Added "nordic:" to commit

ioannisg commented:
> 
> 
> Added "nordic:" to commit

Please, remove Kconfig, though.

"soc: nordic: ....."

Otherwise, good to go

hakonfam commented:
> > Added "nordic:" to commit
> 
> Please, remove Kconfig, though.
> 
> "soc: nordic: ....."
> 
> Otherwise, good to go

fixed

SebastianBoe commented:
@carlescufi : This is ready to be merged.
